### PR TITLE
Upgrade grpc-core to 1.4.0

### PR DESCRIPTION
**Pull request by BusyJay**
This pr upgrade grpc to 1.4.0 and enable static-link by default.

### CODE REVIEW

File: grpc-sys/Cargo.toml [hunk 1]
```diff
@@ -9,10 +9,10 @@ clippy = { version = "*", optional = true }
...
 [build-dependencies]
 gcc = "0.3"
-cmake = { version = "0.1", optional = true }
+cmake = "0.1"
 pkg-config = "0.3"
 
 [features]
 default = []
-static-link = ["cmake"]
+link-sys = []
```
Comment (siddontang) [inline grpc-sys/Cargo.toml]: why rename ?

Comment (BusyJay) [inline grpc-sys/Cargo.toml]: It's not a rename but a new feature.

Comment (siddontang) [inline grpc-sys/Cargo.toml]: But it is still for static build?
btw, we don't need cmake now?

Comment (BusyJay) [inline grpc-sys/Cargo.toml]: No, it's for linking installed library in system.

Comment (BusyJay) [inline grpc-sys/Cargo.toml]: cmake is not an optional deps any more.

**Comment by siddontang** [comment]
LGTM

**Comment by overvenus** [comment]
> enable static-link by default

How to disable static link?

**Comment by BusyJay** [comment]
Enable feature `link-sys`.

**Pull request merged**